state-reconstruct-fetcher/src/types/common.rs

```diff
@@ -85,6 +85,67 @@ impl TryFrom<&abi::Token> for ExtractedToken {
 }
 
 // TODO: Move these to a dedicated parser struct.
+pub fn parse_compressed_state_diffs(
+    bytes: &[u8],
+    pointer: &mut usize,
+    end_bytes_to_ignore: usize,
+) -> Result<Vec<L2ToL1Pubdata>, ParseError> {
+    let mut state_diffs = Vec::new();
+    // Parse the header.
+    let version = u8::from_be_bytes(read_next_n_bytes(bytes, pointer));
+    if version != 1 {
+        return Err(ParseError::InvalidCompressedValue(String::from(
+            "VersionMismatch",
+        )));
+    }
+
+    if *pointer >= bytes.len() {
```

Comment:
isn't this incorrect for end_bytes_to_ignore > 0 ? IMHO the interface is just too complicated: the function should simply take a slice...